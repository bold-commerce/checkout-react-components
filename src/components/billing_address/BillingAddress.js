/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import RadioField from '@boldcommerce/stacks-ui/lib/components/radiofield/RadioField';
import Address from '../address/Address';
import './BillingAddress.css';

const BillingAddress = ({
  address,
  sameAsShipping,
  setSameAsShipping,
  dispatch,
  errors,
  countries,
  provinces,
  showPostalCode,
  showProvince,
  provinceLabel,
  submit,
}) => (
  <section className="FieldSet FieldSet--BillingAddress">
    <div className="FieldSet__Header">
      <div className="FieldSet__Heading">Billing address</div>
    </div>
    <div className="FieldSet__Content">
      <RadioField
        label="Same as shipping address"
        name="billing-address"
        className="RadioButton"
        value="SAME_AS_SHIPPING_ADDRESS"
        checked={sameAsShipping}
        onChange={() => {
          setSameAsShipping(true);
        }}
      />
      <RadioField
        label="Use a different billing address"
        name="billing-address"
        className="RadioButton"
        value="DIFFERENT_BILLING_ADDRESS"
        checked={!sameAsShipping}
        onChange={() => setSameAsShipping(false)}
      />
      { sameAsShipping ? null
        : (
          <Address
            address={address}
            dispatch={dispatch}
            errors={errors}
            countries={countries}
            provinces={provinces}
            showPostalCode={showPostalCode}
            showProvince={showProvince}
            provinceLabel={provinceLabel}
            submit={submit}
          />
        )}
    </div>
  </section>
);

BillingAddress.propTypes = {
  address: PropTypes.any,
  sameAsShipping: PropTypes.bool,
  setSameAsShipping: PropTypes.func,
  dispatch: PropTypes.func,
  errors: PropTypes.any,
  countries: PropTypes.any,
  provinces: PropTypes.any,
  showPostalCode: PropTypes.bool,
  showProvince: PropTypes.bool,
  provinceLabel: PropTypes.string,
  submit: PropTypes.func,
};

export default BillingAddress;
