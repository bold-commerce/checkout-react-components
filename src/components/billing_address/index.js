// eslint-disable-next-line import/prefer-default-export
export { default as BillingAddress } from './BillingAddress';
