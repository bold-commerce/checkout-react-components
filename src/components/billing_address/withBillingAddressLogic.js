import React, {
  useCallback, useEffect, useReducer, useState,
} from 'react';
import { debounce } from '../../helpers/debounce';
import useBillingAddress from '../../hooks/useBillingAddress';
import useShippingAddress from '../../hooks/useShippingAddress';
import addressReducer from '../../reducers/addressReducer';

const withBillingAddressLogic = (Component) => {
  const WithBillingAddressLogic = (props) => {
    const {
      billingAddress, billingAddressErrors, billingAddressCountries, submitBillingAddress,
    } = useBillingAddress();

    const { shippingAddress } = useShippingAddress();

    const [sameAsShipping, setSameAsShipping] = useState(true);

    useEffect(() => {
      if (sameAsShipping) {
        submitBillingAddress(shippingAddress);
      }
    }, [sameAsShipping, JSON.stringify(shippingAddress)]);

    const [address, dispatch] = useReducer(addressReducer, billingAddress);
    let provinces = [];
    let showPostalCode = false;
    let showProvince = false;
    let provinceLabel = 'Province';

    if (address.country_code) {
      const countryData = billingAddressCountries.find((countryItem) => countryItem.iso_code === address.country_code);
      // eslint-disable-next-line no-nested-ternary
      provinces = countryData.provinces.sort((a, b) => ((a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0)));
      showPostalCode = countryData.show_postal_code;
      showProvince = countryData.show_province;
      provinceLabel = countryData.province_label;
    }

    const debouncedSubmit = useCallback(debounce((data) => {
      submitBillingAddress(data);
    }, 1000), [submitBillingAddress]);

    const submit = useCallback(() => {
      debouncedSubmit(address);
    }, [debouncedSubmit, JSON.stringify(address)]);

    const updatedProps = {
      ...props,
      address,
      sameAsShipping,
      setSameAsShipping,
      dispatch,
      errors: billingAddressErrors,
      countries: billingAddressCountries,
      provinces,
      showPostalCode,
      showProvince,
      provinceLabel,
      submit,
    };

    // eslint-disable-next-line react/jsx-props-no-spreading
    return <Component {...updatedProps} />;
  };

  return WithBillingAddressLogic;
};

export default withBillingAddressLogic;
