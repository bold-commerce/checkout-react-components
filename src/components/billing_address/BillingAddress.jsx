/* eslint-disable react/forbid-prop-types */
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import RadioField from '@boldcommerce/stacks-ui/lib/components/radiofield/RadioField';
import { Address } from '../address';
import useCountryInfo from '../../hooks/useCountryInfo';
import { useBillingAddress } from '../../hooks';
import './BillingAddress.css';

const BillingAddress = ({
  billingAddress, countryInfo, billingAddressErrors, billingSameAsShipping, submitBillingAddress, setBillingSameAsShipping, onChange,
}) => {
  const [address, setAddress] = useState(billingAddress);
  const {
    countries,
    provinces,
    showProvince,
    showPostalCode,
    provinceLabel,
  } = useCountryInfo(countryInfo, address);

  useEffect(() => {
    if (onChange) {
      onChange(address);
    }
  }, [address]);

  return (
    <section className="FieldSet FieldSet--BillingAddress">
      <div className="FieldSet__Header">
        <div className="FieldSet__Heading">Billing address</div>
      </div>
      <div className="FieldSet__Content">
        <RadioField
          label="Same as shipping address"
          name="billing-address"
          className="RadioButton"
          value="SAME_AS_SHIPPING_ADDRESS"
          checked={billingSameAsShipping}
          onChange={() => setBillingSameAsShipping(true)}
        />
        <RadioField
          label="Use a different billing address"
          name="billing-address"
          className="RadioButton"
          value="DIFFERENT_BILLING_ADDRESS"
          checked={!billingSameAsShipping}
          onChange={() => setBillingSameAsShipping(false)}
        />
        { billingSameAsShipping ? null
          : (
            <Address
              address={address}
              onChange={(data) => setAddress({
                ...address,
                ...data,
              })}
              errors={billingAddressErrors}
              countries={countries}
              provinces={provinces}
              showPostalCode={showPostalCode}
              showProvince={showProvince}
              provinceLabel={provinceLabel}
              submit={() => submitBillingAddress(address)}
            />
          )}
      </div>
    </section>
  );
};

BillingAddress.propTypes = {
  billingAddress: PropTypes.any,
  countryInfo: PropTypes.array,
  billingAddressErrors: PropTypes.object,
  billingSameAsShipping: PropTypes.bool,
  submitBillingAddress: PropTypes.func,
  setBillingSameAsShipping: PropTypes.func,
  onChange: PropTypes.func,
};

const MemoizedBillingAddress = React.memo(BillingAddress);

const BillingAddressContainer = ({ onChange }) => {
  const {
    billingAddress, countryInfo, billingAddressErrors, billingSameAsShipping, submitBillingAddress, setBillingSameAsShipping,
  } = useBillingAddress();

  return (
    <MemoizedBillingAddress
      billingAddress={billingAddress}
      countryInfo={countryInfo}
      billingAddressErrors={billingAddressErrors}
      billingSameAsShipping={billingSameAsShipping}
      onChange={onChange}
      submitBillingAddress={onChange || submitBillingAddress}
      setBillingSameAsShipping={setBillingSameAsShipping}
    />
  );
};

BillingAddressContainer.propTypes = {
  onChange: PropTypes.func,
};

export default BillingAddressContainer;
