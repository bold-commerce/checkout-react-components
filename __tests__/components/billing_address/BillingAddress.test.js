import React from 'react';
import { render } from '@testing-library/react';
import { BillingAddress } from '../../../src/components/billing_address/BillingAddress';
import {
  addressErrors, countries, emptyAddress, exampleAddress,
} from '../../utils/addressHelpers';

describe('BillingAddress', () => {
  test('renders BillingAddress component', () => {
    const { asFragment } = render(
      <BillingAddress
        billingAddress={emptyAddress}
        countryInfo={countries}
        billingAddressErrors={null}
        billingSameAsShipping={false}
        submitBillingAddress={null}
        setBillingSameAsShipping={null}
        onChange={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders BillingAddress component with errors', () => {
    const { asFragment } = render(
      <BillingAddress
        billingAddress={emptyAddress}
        countryInfo={countries}
        billingAddressErrors={addressErrors}
        billingSameAsShipping={false}
        submitBillingAddress={null}
        setBillingSameAsShipping={null}
        onChange={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders BillingAddress component with pre-filled data', () => {
    const { asFragment } = render(
      <BillingAddress
        billingAddress={exampleAddress}
        countryInfo={countries}
        billingAddressErrors={null}
        billingSameAsShipping={false}
        submitBillingAddress={null}
        setBillingSameAsShipping={null}
        onChange={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders BillingAddress component with same as shipping selected', () => {
    const { asFragment } = render(
      <BillingAddress
        billingAddress={null}
        countryInfo={countries}
        billingAddressErrors={null}
        billingSameAsShipping
        submitBillingAddress={null}
        setBillingSameAsShipping={null}
        onChange={null}
        requiredAddressFields={null}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders BillingAddress component with required fields', () => {
    const requiredFields = [
      'first_name',
      'last_name',
      'address_line_1',
      'address_line_2',
      'business_name',
      'phone_number',
    ];

    const { asFragment } = render(
      <BillingAddress
        billingAddress={null}
        countryInfo={countries}
        billingAddressErrors={null}
        billingSameAsShipping
        submitBillingAddress={null}
        setBillingSameAsShipping={null}
        onChange={null}
        requiredAddressFields={requiredFields}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
