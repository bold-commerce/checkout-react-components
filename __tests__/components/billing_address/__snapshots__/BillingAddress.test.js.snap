// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingAddress renders BillingAddress component 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--BillingAddress"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Billing address
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio-field__radio"
          >
            <input
              aria-checked="false"
              class="stx-radio__input"
              id="stxField1"
              name="billing-address"
              role="radio"
              type="radio"
              value="SAME_AS_SHIPPING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField1"
          >
            Same as shipping address
          </label>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio--is-checked stx-radio-field__radio"
          >
            <input
              aria-checked="true"
              checked=""
              class="stx-radio__input"
              id="stxField2"
              name="billing-address"
              role="radio"
              type="radio"
              value="DIFFERENT_BILLING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField2"
          >
            Use a different billing address
          </label>
        </div>
      </div>
      <div
        class="FieldSet--Address"
      >
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--FirstName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField3"
                name="first_name"
                placeholder="First name (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--LastName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField4"
                name="last_name"
                placeholder="Last name (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Company"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField5"
                name="business_name"
                placeholder="Company (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Address"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField6"
                name="address_line_1"
                placeholder="Address (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--Address2"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField7"
                name="address_line_2"
                placeholder="Apt, suite, etc."
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--City"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField8"
                name="city"
                placeholder="City (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-select SelectField Field--Country"
          >
            <div
              class="stx-select stx-select--with-placeholder"
            >
              <select
                class="stx-select__select-element"
                id="stxField9"
              >
                <option
                  hidden=""
                  value="___PLACEHOLDER___"
                >
                  Select a country
                </option>
                <option
                  value="CA"
                >
                  Canada
                </option>
                <option
                  value="US"
                >
                  United States
                </option>
              </select>
              <span
                class="stx-select__icon"
              >
                <svg
                  class="stx-select-icon__icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="stx-select-icon__path"
                    d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                    fill="#6B6B6B"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Phone"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField10"
                name="phone_number"
                placeholder="Phone (optional)"
                type="tel"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`BillingAddress renders BillingAddress component with errors 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--BillingAddress"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Billing address
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio-field__radio"
          >
            <input
              aria-checked="false"
              class="stx-radio__input"
              id="stxField11"
              name="billing-address"
              role="radio"
              type="radio"
              value="SAME_AS_SHIPPING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField11"
          >
            Same as shipping address
          </label>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio--is-checked stx-radio-field__radio"
          >
            <input
              aria-checked="true"
              checked=""
              class="stx-radio__input"
              id="stxField12"
              name="billing-address"
              role="radio"
              type="radio"
              value="DIFFERENT_BILLING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField12"
          >
            Use a different billing address
          </label>
        </div>
      </div>
      <div
        class="FieldSet--Address"
      >
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--alert stx-field--with-input Field Field--FirstName"
          >
            <div
              class="stx-input stx-input--alert"
            >
              <input
                class="stx-input__input-element"
                id="stxField13"
                name="first_name"
                placeholder="First name (optional)"
                type="text"
                value=""
              />
            </div>
            <div
              class="stx-field__message"
            >
              Enter a first name
            </div>
          </div>
          <div
            class="stx-field stx-field--alert stx-field--with-input Field Field--LastName"
          >
            <div
              class="stx-input stx-input--alert"
            >
              <input
                class="stx-input__input-element"
                id="stxField14"
                name="last_name"
                placeholder="Last name (optional)"
                type="text"
                value=""
              />
            </div>
            <div
              class="stx-field__message"
            >
              Enter a last name
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--alert stx-field--with-input Field Field--Company"
          >
            <div
              class="stx-input stx-input--alert"
            >
              <input
                class="stx-input__input-element"
                id="stxField15"
                name="business_name"
                placeholder="Company (optional)"
                type="text"
                value=""
              />
            </div>
            <div
              class="stx-field__message"
            >
              Enter a company name
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Address"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField16"
                name="address_line_1"
                placeholder="Address (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--Address2"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField17"
                name="address_line_2"
                placeholder="Apt, suite, etc."
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--City"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField18"
                name="city"
                placeholder="City (optional)"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--alert stx-field--with-select SelectField Field--Country"
          >
            <div
              class="stx-select stx-select--with-placeholder stx-select--alert"
            >
              <select
                class="stx-select__select-element"
                id="stxField19"
              >
                <option
                  hidden=""
                  value="___PLACEHOLDER___"
                >
                  Select a country
                </option>
                <option
                  value="CA"
                >
                  Canada
                </option>
                <option
                  value="US"
                >
                  United States
                </option>
              </select>
              <span
                class="stx-select__icon"
              >
                <svg
                  class="stx-select-icon__icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="stx-select-icon__path"
                    d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                    fill="#6B6B6B"
                  />
                </svg>
              </span>
            </div>
            <div
              class="stx-field__message"
            >
              Select a country
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--alert stx-field--with-input Field Field--Phone"
          >
            <div
              class="stx-input stx-input--alert"
            >
              <input
                class="stx-input__input-element"
                id="stxField20"
                name="phone_number"
                placeholder="Phone (optional)"
                type="tel"
                value=""
              />
            </div>
            <div
              class="stx-field__message"
            >
              Enter a phone number
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`BillingAddress renders BillingAddress component with pre-filled data 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--BillingAddress"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Billing address
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio-field__radio"
          >
            <input
              aria-checked="false"
              class="stx-radio__input"
              id="stxField21"
              name="billing-address"
              role="radio"
              type="radio"
              value="SAME_AS_SHIPPING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField21"
          >
            Same as shipping address
          </label>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio--is-checked stx-radio-field__radio"
          >
            <input
              aria-checked="true"
              checked=""
              class="stx-radio__input"
              id="stxField22"
              name="billing-address"
              role="radio"
              type="radio"
              value="DIFFERENT_BILLING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField22"
          >
            Use a different billing address
          </label>
        </div>
      </div>
      <div
        class="FieldSet--Address"
      >
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--FirstName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField23"
                name="first_name"
                placeholder="First name (optional)"
                type="text"
                value="John"
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--LastName"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField24"
                name="last_name"
                placeholder="Last name (optional)"
                type="text"
                value="Doe"
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Company"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField25"
                name="business_name"
                placeholder="Company (optional)"
                type="text"
                value="Bold Commerce"
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Address"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField26"
                name="address_line_1"
                placeholder="Address (optional)"
                type="text"
                value="123 Fake Street"
              />
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--Address2"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField27"
                name="address_line_2"
                placeholder="Apt, suite, etc."
                type="text"
                value="Unit 12"
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--City"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField28"
                name="city"
                placeholder="City (optional)"
                type="text"
                value="Winnipeg"
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-select SelectField Field--Country"
          >
            <div
              class="stx-select stx-select--with-placeholder"
            >
              <select
                class="stx-select__select-element"
                id="stxField29"
              >
                <option
                  hidden=""
                  value="___PLACEHOLDER___"
                >
                  Select a country
                </option>
                <option
                  value="CA"
                >
                  Canada
                </option>
                <option
                  value="US"
                >
                  United States
                </option>
              </select>
              <span
                class="stx-select__icon"
              >
                <svg
                  class="stx-select-icon__icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="stx-select-icon__path"
                    d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                    fill="#6B6B6B"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="stx-field stx-field--with-select SelectField Field--Province"
          >
            <div
              class="stx-select stx-select--with-placeholder"
            >
              <select
                class="stx-select__select-element"
                id="stxField31"
              >
                <option
                  hidden=""
                  value="___PLACEHOLDER___"
                >
                  Select a province
                </option>
                <option
                  value="AB"
                >
                  Alberta
                </option>
                <option
                  value="BC"
                >
                  British Columbia
                </option>
                <option
                  value="MB"
                >
                  Manitoba
                </option>
                <option
                  value="NB"
                >
                  New Brunswick
                </option>
                <option
                  value="NL"
                >
                  Newfoundland and Labrador
                </option>
                <option
                  value="NT"
                >
                  Northwest Territories
                </option>
                <option
                  value="NS"
                >
                  Nova Scotia
                </option>
                <option
                  value="NU"
                >
                  Nunavut
                </option>
                <option
                  value="ON"
                >
                  Ontario
                </option>
                <option
                  value="PE"
                >
                  Prince Edward Island
                </option>
                <option
                  value="QC"
                >
                  Quebec
                </option>
                <option
                  value="SK"
                >
                  Saskatchewan
                </option>
                <option
                  value="YT"
                >
                  Yukon Territory
                </option>
              </select>
              <span
                class="stx-select__icon"
              >
                <svg
                  class="stx-select-icon__icon"
                  viewBox="0 0 24 24"
                >
                  <path
                    class="stx-select-icon__path"
                    d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                    fill="#6B6B6B"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="stx-field stx-field--with-input Field Field--Postal_Code"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField32"
                name="postal"
                placeholder="Postal code / ZIP"
                type="text"
                value="R3A1B3"
              />
            </div>
          </div>
        </div>
        <div
          class="FieldGroup"
        >
          <div
            class="stx-field stx-field--with-input Field Field--Phone"
          >
            <div
              class="stx-input"
            >
              <input
                class="stx-input__input-element"
                id="stxField30"
                name="phone_number"
                placeholder="Phone (optional)"
                type="tel"
                value="204-123-4567"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`BillingAddress renders BillingAddress component with required fields 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--BillingAddress"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Billing address
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio--is-checked stx-radio-field__radio"
          >
            <input
              aria-checked="true"
              checked=""
              class="stx-radio__input"
              id="stxField35"
              name="billing-address"
              role="radio"
              type="radio"
              value="SAME_AS_SHIPPING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField35"
          >
            Same as shipping address
          </label>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio-field__radio"
          >
            <input
              aria-checked="false"
              class="stx-radio__input"
              id="stxField36"
              name="billing-address"
              role="radio"
              type="radio"
              value="DIFFERENT_BILLING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField36"
          >
            Use a different billing address
          </label>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`BillingAddress renders BillingAddress component with same as shipping selected 1`] = `
<DocumentFragment>
  <section
    class="FieldSet FieldSet--BillingAddress"
  >
    <div
      class="FieldSet__Header"
    >
      <div
        class="FieldSet__Heading"
      >
        Billing address
      </div>
    </div>
    <div
      class="FieldSet__Content"
    >
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio--is-checked stx-radio-field__radio"
          >
            <input
              aria-checked="true"
              checked=""
              class="stx-radio__input"
              id="stxField33"
              name="billing-address"
              role="radio"
              type="radio"
              value="SAME_AS_SHIPPING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField33"
          >
            Same as shipping address
          </label>
        </div>
      </div>
      <div
        class="stx-field stx-field--with-radio-field RadioButton"
      >
        <div
          class="stx-radio-field"
        >
          <span
            class="stx-radio stx-radio-field__radio"
          >
            <input
              aria-checked="false"
              class="stx-radio__input"
              id="stxField34"
              name="billing-address"
              role="radio"
              type="radio"
              value="DIFFERENT_BILLING_ADDRESS"
            />
            <span
              class="stx-radio__box"
            />
          </span>
          <label
            class="stx-radio-field__label"
            for="stxField34"
          >
            Use a different billing address
          </label>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
